### PR TITLE
PF-2692: add a flagsmith service

### PR DESCRIPTION
**Pull request by yuhuyoyo**
A small wrapper around the java library

### CODE REVIEW

File: src/test/resources/application-flagsmith-test.properties [hunk 1]
```diff
@@ -0,0 +1,3 @@
+terra.common.flagsmith.enabled=true
+terra.common.flagsmith.server-side-api-key=ser.Vn8MpGpej26aShW7CXsEaR
+terra.common.flagsmith.api-url=https://terra-flagsmith.api.verily.com/api/v1/
```
Comment (zloery) [inline src/test/resources/application-flagsmith-test.properties]: Are these ok to check into terra-common-lib? And is it ok for this upstream repo to call the internal-env instance?

Comment (yuhuyoyo) [reply src/test/resources/application-flagsmith-test.properties]: i created a test env but now I moved it to use the public one.

**Review by zloery — APPROVED** [review]

File: src/main/java/bio/terra/common/flagsmith/FlagsmithService.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+public class FlagsmithService {
+  private static final Logger LOGGER = LoggerFactory.getLogger(FlagsmithService.class);
+
+  @Autowired private final FlagsmithProperties flagsmithProperties;
+
+  FlagsmithService(FlagsmithProperties flagsmithProperties) {
+    this.flagsmithProperties = flagsmithProperties;
+  }
+
+  public boolean isFeatureEnabled(String feature, Optional<Boolean> defaultValue) {
```
Comment (dexamundsen) [inline src/main/java/bio/terra/common/flagsmith/FlagsmithService.java]: Minor: consider this signature

public Optional<Boolean> isFeatureEnabled(String feature) {
  // if flagsmith is not enabled, optional.emtpy()
}

reason : this library does not need to know the default value. Default behavior is determined by the client. 

Alternate - throw an exception if flagsmith is not set up.

**Review by dexamundsen — APPROVED** [review]

File: src/main/java/bio/terra/common/flagsmith/FlagsmithService.java (hunk 2 above)
Comment (yuhuyoyo) [reply src/main/java/bio/terra/common/flagsmith/FlagsmithService.java]: good idea.

**Pull request merged**